docs/development/linux-bridge-primary-nic-scenario.md

```diff
@@ -0,0 +1,787 @@
+  literal: PATH(4)
+- pos: 54
+  type: IDENTITY(1)
+  literal: routes
+- pos: 60
+  type: PATH(4)
+  literal: PATH(4)
+- pos: 61
+  type: IDENTITY(1)
+  literal: running
+- pos: 68
+  type: PATH(4)
+  literal: PATH(4)
+- pos: 69
+  type: NUMBER(2)
+  literal: "0"
+- pos: 70
+  type: PATH(4)
+  literal: PATH(4)
+- pos: 71
+  type: IDENTITY(1)
+  literal: next-hop-interface
+- pos: 88
+  type: EOF(0)
+  literal: ""
+
+```
+
+**AST**
+```yaml
+pos: 0
+equal:
+  - pos: 0
+    path:
+      - pos: 0
+        id: routes
+      - pos: 7
+        id: running
+      - pos: 15
+        id: next-hop-interface
+  - pos: 35
+    path:
+      - pos: 35
+        id: capture
+      - pos: 43
+        id: default-gw
+      - pos: 54
+        id: routes
+      - pos: 61
+        id: running
+      - pos: 69
+        idx: 0
+      - pos: 71
+        id: next-hop-interface
+
+```
+#### base-iface: interfaces.name==capture.default-gw.routes.running.0.next-hop-interface
+
+**Tokens**
+```yaml
+- pos: 0
+  type: IDENTITY(1)
+  literal: interfaces
+- pos: 10
+  type: PATH(4)
+  literal: PATH(4)
+- pos: 11
+  type: IDENTITY(1)
+  literal: name
+- pos: 15
+  type: EQUALITY(10)
+  literal: EQUALITY(10)
+- pos: 17
+  type: IDENTITY(1)
+  literal: capture
+- pos: 24
+  type: PATH(4)
+  literal: PATH(4)
+- pos: 25
+  type: IDENTITY(1)
+  literal: default-gw
+- pos: 35
+  type: PATH(4)
+  literal: PATH(4)
+- pos: 36
+  type: IDENTITY(1)
+  literal: routes
+- pos: 42
+  type: PATH(4)
+  literal: PATH(4)
+- pos: 43
+  type: IDENTITY(1)
+  literal: running
+- pos: 50
+  type: PATH(4)
+  literal: PATH(4)
+- pos: 51
+  type: NUMBER(2)
+  literal: "0"
+- pos: 52
+  type: PATH(4)
+  literal: PATH(4)
+- pos: 53
+  type: IDENTITY(1)
+  literal: next-hop-interface
+- pos: 70
+  type: EOF(0)
+  literal: ""
+
+```
+
+**AST**
+```yaml
+pos: 0
+equal:
+  - pos: 0
+    path:
+      - pos: 0
+        id: interfaces
+      - pos: 11
+        id: name
+  - pos: 17
+    path:
+      - pos: 17
+        id: capture
+      - pos: 25
+        id: default-gw
+      - pos: 36
+        id: routes
+      - pos: 43
+        id: running
+      - pos: 51
+        idx: 0
+      - pos: 53
+        id: next-hop-interface
+
+```
+#### bridge-routes: capture.base-iface-routes | routes.running.next-hop-interface="br1"
+
+**Tokens**
+```yaml
+- pos: 0
+  type: IDENTITY(1)
+  literal: capture
+- pos: 7
+  type: PATH(4)
+  literal: PATH(4)
+- pos: 8
+  type: IDENTITY(1)
+  literal: base-iface-routes
+- pos: 26
+  type: PIPE(5)
+  literal: PIPE(5)
+- pos: 28
+  type: IDENTITY(1)
+  literal: routes
+- pos: 34
+  type: PATH(4)
+  literal: PATH(4)
+- pos: 35
+  type: IDENTITY(1)
+  literal: running
+- pos: 42
+  type: PATH(4)
+  literal: PATH(4)
+- pos: 43
+  type: IDENTITY(1)
+  literal: next-hop-interface
+- pos: 61
+  type: ASSIGN(9)
+  literal: ASSIGN(9)
+- pos: 62
+  type: STRING(3)
+  literal: br1
+- pos: 66
+  type: EOF(0)
+  literal: ""
+- pos: 66
+  type: EOF(0)
+  literal: ""
+
+```
+
+**AST**
```

Comment:
I don't understand what is the root of this tree. It would be much easier to follow the basic idea if the AST will be a diagram of a tree and not a yaml.